src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1220,49 +1223,275 @@ static bool addTriggeredSubscriptions
+  std::string inRegex      = "{ $in: [ " + spathRegex + ", null ] }";
+  BSONObj     spBson       = fromjson(inRegex);
+
+  /* Note the $or on entityType, to take into account matching in subscriptions with no entity type */
+  BSONObj queryNoPattern = BSON(
+                entIdQ << entityId <<
+                "$or" << BSON_ARRAY(
+                    BSON(entTypeQ << entityType) <<
+                    BSON(entTypeQ << BSON("$exists" << false))) <<
+                entPatternQ << "false" <<
+                condTypeQ << ON_CHANGE_CONDITION <<
+                condValueQ << attr <<
+                CSUB_EXPIRATION   << BSON("$gt" << (long long) getCurrentTime()) <<
+                CSUB_SERVICE_PATH << spBson);
+
+  /* This is JavaScript code that runs in MongoDB engine. As far as I know, this is the only
+   * way to do a "reverse regex" query in MongoDB (see
+   * http://stackoverflow.com/questions/15966991/mongodb-reverse-regex/15989520).
+   * Note that although we are using a isPattern=true in the MongoDB query besides $where, we
+   * also need to check that in the if statement in the JavaScript function given that a given
+   * sub document could include both isPattern=true and isPattern=false documents */
+  std::string function = std::string("function()") +
+         "{" +
+            "for (var i=0; i < this."+CSUB_ENTITIES+".length; i++) {" +
+                "if (this."+CSUB_ENTITIES+"[i]."+CSUB_ENTITY_ISPATTERN+" == \"true\" && " +
+                    "(this."+CSUB_ENTITIES+"[i]."+CSUB_ENTITY_TYPE+" == \""+entityType+"\" || " +
+                        "this."+CSUB_ENTITIES+"[i]."+CSUB_ENTITY_TYPE+" == \"\" || " +
+                        "!(\""+CSUB_ENTITY_TYPE+"\" in this."+CSUB_ENTITIES+"[i])) && " +
+                    "\""+entityId+"\".match(this."+CSUB_ENTITIES+"[i]."+CSUB_ENTITY_ID+")) {" +
+                    "return true; " +
+                "}" +
+            "}" +
+            "return false; " +
+         "}";
+  LM_T(LmtMongo, ("JS function: %s", function.c_str()));
+
+  BSONObjBuilder  queryPattern;
+
+  queryPattern.append(entPatternQ, "true");
+  queryPattern.append(condTypeQ, ON_CHANGE_CONDITION);
+  queryPattern.append(condValueQ, attr);
+  queryPattern.append(CSUB_EXPIRATION, BSON("$gt" << (long long) getCurrentTime()));
+  queryPattern.append(CSUB_SERVICE_PATH, spBson);
+  queryPattern.appendCode("$where", function);
+
+  // FIXME: condTypeQ, condValueQ and servicePath part could be "factorized" out of the $or clause
+  BSONObj query = BSON("$or" << BSON_ARRAY(queryNoPattern << queryPattern.obj()));
+
+  LM_T(LmtMongo, ("query() in '%s' collection: '%s'",
+                  getSubscribeContextCollectionName(tenant).c_str(),
+                  query.toString().c_str()));
+
+  /* Do the query */
+  auto_ptr<DBClientCursor> cursor;
+  try
+  {
+    connection      = getMongoConnection();
+    cursor = connection->query(getSubscribeContextCollectionName(tenant).c_str(), query);
```

Comment:
This is no longer the way of accesing DB... Please have a look to the methods in the connectionOperations.cpp/h